aiohttp/helpers.py

```diff
@@ -733,3 +733,35 @@ def content_length(self, *, _CONTENT_LENGTH=hdrs.CONTENT_LENGTH):
 
         if content_length:
             return int(content_length)
+
+
+def format_parameter_value(value):
+    """Wrap a parameter value in quotes, if necessary.
+
+    Expects and returns a byte-string.
+    """
+    # Refer to RFCs 7231, 7230, 5234.
+    #
+    # parameter      = token "=" ( token / quoted-string )
+    # token          = 1*tchar
+    # quoted-string  = DQUOTE *( qdtext / quoted-pair ) DQUOTE
+    # qdtext         = HTAB / SP / %x21 / %x23-5B / %x5D-7E / obs-text
+    # obs-text       = %x80-FF
+    # quoted-pair    = "\" ( HTAB / SP / VCHAR / obs-text )
+    # tchar          = "!" / "#" / "$" / "%" / "&" / "'" / "*"
+    #                  / "+" / "-" / "." / "^" / "_" / "`" / "|" / "~"
+    #                  / DIGIT / ALPHA
+    #                  ; any VCHAR, except delimiters
+    # VCHAR           = %x21-7E
+    valid_tchar_regex = br"^[!#$%&'*+\-.^_`|~\w]+$"
+    if re.match(valid_tchar_regex, value):
```

Comment:
Please use compiled regular exception

Reply:
Could you explain the rationale for this?

Reply:
Compiled regexs are faster, isn't it?

Reply:
Yes, but only if they are cached properly so they aren't recompiled every time. Python does some internal caching for regular expressions already, so unless we need a lot of regular expressions, the internal cache should cover most use cases.

In fact, if we don't cache the regex and compile it every time we need it, we actually degrade performance compared to having Python (or the re module) handle caching automatically.

Reply:
I propose to cache regexps at module level, like [here](https://github.com/aio-libs/aiohttp/blob/master/aiohttp/helpers.py#L452
`re` module has 512 slots for LRU cache of compiled regexps.
Thus the program may degrade performance if other code uses non-complied exceptions for solving own tasks. If the total regexps number exceeds 512 -- `re` will recompile them constantly.

Reply:
This function will only be run once for every multipart document sent, so I believe the performance impact will be minimal at best, but we can do it anyway if that's your standard.

Reply:
Thanks.
1. It is the aiohttp standard.
2. The function runs once per multipart document but for high-load site it could clash with other handlers and their functionality.